Docs improvement

obrusvit opened the pull request:
<!--
If you are a core dev:

Don't forget to set up the fields:
- assign yourself
- set priority to same as original issue
- add PR to sprint
- if Draft -> "in progress"
- if final PR -> "needs review"

If you're an external contributor, you can ignore this text.
-->
This PR fixes some minor issues and obsolete information in firmware documentation which I stumbled upon during my onboarding. 

In addition, this PR adds a small script which lists markdown files present in `docs/` which are not referenced in `docs/SUMMARY.md` - this is an effort to prevent a situation we had where I didn't know about "Hello World" tutorial because I didn't see it in the web version of the documentation. I referenced the "Hello World" tutorial now so I also rebased it on top of current `main`. The script currently lists:

- `core/misc/disc1.md`
- `storage/addon-block.md`
- `python/show-text.md`

Hannsek commented on docs/core/src/apps.md:
```diff
@@ -2,15 +2,34 @@
 The folder `src/apps/` is the place where all the user-facing features are implemented.
 
-Each app has a `boot()` function in the module's \_\_init\_\_ file. This functions assigns what function should be called if some specific message was received. In other words, it is a link between the MicroPython functions and the Protobuf messages.
+Each app must be registered by the  `register` function inside the file `workflow_handlers.py`. This functions assigns what function should be called if some specific message was received. In other words, it is a link between the MicroPython functions and the Protobuf messages.
```
obsolete space right before `register`

matejcik commented on docs/core/index.md:
```diff
@@ -1,5 +1,13 @@
+
+Trezor Core uses [MicroPython](https://github.com/micropython/micropython), it is a Python implementation for embedded systems, which allows us to have an application layer in Python, which makes the code significantly more readable and sustainable. This is what you find in the `src` folder.
+
+Not everything is in Python though, we need to use C occasionally, usually for performance reasons. That is what `embed/extmod` is for. It extends MicroPython's modules with a number of our owns and serves as a bridge between C and Python codebase. Related to that, `mocks` contain Python mocks of those functions to improve readability and IDE functioning.
```
add a couple words about Rust here?

obrusvit replied:
[1dbb487](https://github.com/trezor/trezor-firmware/pull/3515/commits/1dbb48750d4878a6e029788c62d23e40a411f009)

matejcik reviewed (commented):
`check_docs_summary` should be added to one of the prebuild check CI steps. `style_check` most likely?

obrusvit commented:
> `check_docs_summary` should be added to one of the prebuild check CI steps. `style_check` most likely?
Agreed, added it to Makefile: [0b83095](https://github.com/trezor/trezor-firmware/pull/3515/commits/0b8309565b424a6379335e77d61bcd2e7d2974a8) 

However, I haven't yet linked the 3 missing files, they might be obsolete.

matejcik commented:
* show-text.md should be dropped
* addon-block.md should be dropped (that's actually a very old design for the translation blob)
* disc1.md should be linked

obrusvit commented:
> > Agreed, added it to Makefile: [0b83095](https://github.com/trezor/trezor-firmware/commit/0b8309565b424a6379335e77d61bcd2e7d2974a8)
> 
> that doesn't help if it does not _fail_ the CI step though. nobody reads the warnings no a green test

Ok. Now it fails loudly
![image](https://github.com/trezor/trezor-firmware/assets/14001709/e9fffbdc-745d-4981-9dfe-ec49ea950a15)

And I removed the obsolete docs.
